vignettes/compareScenariosRemind2.Rmd

```diff
@@ -0,0 +1,199 @@
+
+
+# Viewing variables available in mif Files
+
+The variables in the reported mif-files (`remind/output/<run_folder>/REMIND_generic_<run>.mif`) have a hierarchical structure induced by the character `|`. This structure can be viewed in a hierarchical list.
+
+## Via output script in REMIND
+
+In a remind folder with finished runs (with mif-files):
+
+1. execute `Rscript output.R`
+2. choose `Comparison across runs`
+3. choose `varListHtml`
+4. choose the runs you want to read the mif files from
+5. choose a file name prefix if you like
+
+Then the creation of the a HTML-file containing the hierarchical list of variables names in the chosen mif files is started. Note that it does not use slurm at the moment.
+
+## Via package function in remind2
+
+See `?remind2::variablesAsList`.
+
+If connected to the PIK network (possibly via VPN) execute the following code to get an overview of the variables available in the automated model test mifs.
+
+```{r}
+library(remind2)
+loadModeltest()
+varList <- variablesAsList(data, entry = "INFO")
+View(varList) # in RStudio
+```
+
+See also `?createVarListHtml` to create an HTML-file containing this info.
+
+# How to add a new plot to remind2
+
+1. If not done already, fork <https://github.com/pik-piam/remind2>.
+2. In your personal github remind2 repository, create a new branch.
+3. Clone the repository and switch to the newly created branch to get your local copy.
+4. In your local copy, open `remind2.Rproj` in RStudio.
+5. Press `CTRL+SHIFT+L` to call `devtools::load_all(".")`, which loads `remind2` from your local copy.
+6. If it is sufficient to test your new plots on the latest AMTs and you are connected to the PIK network (possibly via VPN), call `loadModeltest(folder = "some/folder")`. Otherwise call an adapted form of 
+
+```{r}
+loadCs2Data(
+  c("path/to/scen1.mif", "path/to/scen2.mif"), # TODO.
+  "path/to/historical.mif") # TODO.
+```
+
+This might take some time (up to 1min). After this, an R-object `data` should be available. Test this by typing `data` into the RStudio Console. The first rows of a *tibble* containing some REMIND data should appear. If a bunch of R-code appears, something went wrong.
+
+7. Open the section-Rmd-file in which you want to add the plot. The files are in the folder `inst/compareScenarios/` and have names of the form `cs_NN_XXXX.Rmd`, where `NN` is a two digit number and `XXXX` resembles the section title.
+8. You should be able to execute any chunk in the section Rmd by clicking the play button in its top right corner.
+9. Insert a new chunk by copying an old one or by pressing `CTRL+ALT+I`. Note: It is better to not assign names to the chunks as cs2 will crash if you used the same name twice.
```

Comment:
But most chunks have names, see https://github.com/pik-piam/remind2/blob/master/inst/markdown/compareScenarios2/cs2_02_macro.Rmd

Reply:
I know, this text was written for the vignette by Christof when creating compareScenarios2 and I migrated all the parts that are not outdated to the new vignette. In practice, they did not stick to this convention. I think, it is meant as a warning for users when introducing new chunks so they are aware that this might be a source of compilation errors. We could rephrase it to make that clearer, but I think we want to keep the existing chunk names.